cli/src/main/kotlin/com/meamoria/lexurgy/cli/server/Server.kt

```diff
@@ -39,36 +63,64 @@ data class StringCollector(val strings: MutableList<String> = mutableListOf()) :
     }
 }
 
-fun runServer(changes: Path) {
-    val changer = soundChangerFromLscFile(changes)
+fun applyChanges(changer: SoundChanger?, request: ApplyChangesRequest) {
+    if (changer == null) throw Exception("No sound changer loaded.")
+    val collector = StringCollector()
+    val intermediates = changer.changeWithIntermediates(
+        request.words,
+        request.startAt,
+        request.stopBefore,
+        request.traceWords,
+        request.romanize,
+        collector
+    )
+
+    val words = intermediates[null]!!
+    val trueIntermediates: Map<String, List<String>> =
+        intermediates.filterKeys { it != null }.mapKeys { it.key!! }
 
+    printResponse(ServerResponse.Changed(words, trueIntermediates, collector.strings))
+}
+
+fun printResponse(response: ServerResponse) {
+    println(Json.encodeToString(ServerResponse.serializer(), response))
+}
+
+fun runServer(changes: Path?) {
+    var changer = changes?.let { soundChangerFromLscFile(it) }
     while (true) {
         try {
             val line = readlnOrNull() ?: break
-            val request = Json.decodeFromString<ServerRequest>(line)
-            val collector = StringCollector()
-            val intermediates = changer.changeWithIntermediates(
-                request.words,
-                request.startAt,
-                request.stopBefore,
-                request.traceWords,
-                request.romanize,
-                collector
-            )
-            val words = intermediates[null]!!
-
-            val trueIntermediates: Map<String, List<String>> =
-                intermediates.filterKeys { it != null }.mapKeys { it.key!! }
-            println(
-                Json.encodeToString(
-                    ServerResponse.serializer(),
-                    ServerResponse.Changed(words, trueIntermediates, collector.strings)
+            val request = Json.parseToJsonElement(line)
+
+            // For backwards-compatibility, allow the request type to be omitted. We
+            // default to a request type of APPLY_CHANGES in that case.
+            // TODO: Consider emitting a deprecation warning for this?
```

Comment:
I don't think the old form should be deprecated. We can discuss, but please remember to remove the TODO